### PR TITLE
Feature/refactor

### PR DESCRIPTION
**Pull request by asariev**

### CODE REVIEW

File: src/main/java/io/onroad/eventconsumer/domain/mapper/LoginViewMapper.java [hunk 1]
```diff
@@ -8,10 +8,11 @@
 @Component
 public class LoginViewMapper {
 
-  public LoginView toUserView(String token) {
+  public LoginView toUserView(String token, User user) {
 
     var loginView = new LoginView();
     loginView.setJwtToken(token);
+    loginView.setId(String.valueOf(user.getId()));
```
Comment (onroadeld) [inline src/main/java/io/onroad/eventconsumer/domain/mapper/LoginViewMapper.java:15]: Why we need id during login?

File: src/main/java/io/onroad/eventconsumer/domain/dto/ProfileRequest.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package io.onroad.eventconsumer.domain.dto;
+
+import lombok.Data;
+
+@Data
+public class ProfileRequest {
+
+    private String id;
```
Comment (onroadeld) [inline src/main/java/io/onroad/eventconsumer/domain/dto/ProfileRequest.java:8]: Let's use this API definition?

https://github.com/onroadeld/event-consumer/blob/main/docs/api.yml#L78

File: src/main/java/io/onroad/eventconsumer/api/DriverApi.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+@Tag(name = "Users")
+@RestController
+@RequestMapping(path = "api/user")
+@RolesAllowed({Role.SUPER_ADMIN, Role.MOTOR_CARRIER_SUPPORT_PERSONNEL, Role.DRIVER})
+@RequiredArgsConstructor
+public class DriverApi {
+
+    private final UserService userService;
+
+    @PostMapping("/profile")
```
Comment (onroadeld) [inline src/main/java/io/onroad/eventconsumer/api/DriverApi.java:30]: We could use this API definition for profile API

https://github.com/onroadeld/event-consumer/blob/main/docs/api.yml#L78

File: src/main/java/io/onroad/eventconsumer/domain/mapper/LoginViewMapper.java (hunk 1 above)
Comment (asariev) [reply src/main/java/io/onroad/eventconsumer/domain/mapper/LoginViewMapper.java:15]: For mobile, they can hash two users with id, Mobile should keep two tokens at the same time for co-driver

Comment (onroadeld) [reply src/main/java/io/onroad/eventconsumer/domain/mapper/LoginViewMapper.java]: Dos, they should use the `/api/me` endpoint to fetch user related information or they can get the unique id from jwt token. JWT token can be easily decoded.

I have decoded this token

eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiIxLHN1cGVyLmFkbWluQG9ucm9hZC5pbyIsImlzcyI6Im9ucm9hZC5pbyIsImlhdCI6MTYyNzIyNzEwOCwiZXhwIjoxNjI3ODMxOTA4fQ.yrLqJIO4vKrYY5UEG1DwpGxz8Z4dLSGyjDYe_nvjoTciJXPVYeWS3bkFJPTxFt7aHW4sfD35Q7gGe2XGkvnsUw

Using 

https://jwt.io/

```
{
  "sub": "1,super.admin@onroad.io",
  "iss": "onroad.io",
  "iat": 1627227108,
  "exp": 1627831908
}
```

As you can see sub contains id and email.

File: src/test/java/io/onroad/eventconsumer/api/TestDriverApi.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@WithUserDetails(value = "support.personnel@onroad.io")
+@ActiveProfiles("test")
+@Sql(scripts = "classpath:schema.sql")
+@Sql(scripts = "classpath:data.sql")
+public class TestDriverApi {
```
Comment (onroadeld) [inline src/test/java/io/onroad/eventconsumer/api/TestDriverApi.java:30]: This file names have been changed on master. Could you merge it and change accordingly here?

File: src/main/java/io/onroad/eventconsumer/service/UserService.java [hunk 5]
```diff
@@ -102,5 +102,8 @@ public UserView getUser(long id) {
     return userViewMapper.toUserView(userRepo.getById(id));
   }
 
+  public UserView getProfile(User loggedUser) {
+    return userViewMapper.toUserView(loggedUser);
```
Comment (onroadeld) [inline src/main/java/io/onroad/eventconsumer/service/UserService.java:106]: I see that UserView has the following properties

```
userView.setEmail(user.getUsername());
    userView.setFirstName(user.getFirstName());
    userView.setLastName(user.getLastName());
    userView.setDriverLicenseNumber(user.getDriversLicense()!= null ? user.getDriversLicense().getNumber() : "");
    
   Do we need more information on the profile page? For example company names, current duty status etc?
   
   We need to change api.yaml file that matches current implementation or the other way around. 
   
   Also let's remove `coDriver` from the `/api/me` endpoint documentation.

**Review by onroadeld — COMMENTED** [review]
Let's remove if these ones are also not needed

```
              company:
                type: object
                properties:
                  name:
                    type: string
              dutyStatus:
                type: object
                properties:
                  abbreviation:
                    example: OFF
                  dataCoding:
                    example: 1
                  name:
                    example: Off Duty

**Review by onroadeld — CHANGES_REQUESTED** [review]
Also `/api/me` expects user information under user property

```
user:
                type: object
                properties:
                  firstName:
                    type: string
                    example: Tomas
                  lastName:
                    type: string
                    example: Edison
                  email:
                    type: string
                    example: tomas.edison@onroadnetworks.com

**Review by onroadeld — APPROVED** [review]

**Pull request merged**